test/Sketch.hs

```diff
@@ -444,6 +444,28 @@ t7 x y z =
         ((\mbc cd -> maybe (Right Nothing) (\bc -> Left $ fmap ((cd . bc) .)) mbc) <$> y))
         ((&) <$> z)
 
+------------------------ McCarthy's Conditional combinator -------------------------
+-- See: http://www4.di.uminho.pt/~jno/ps/pdbc.pdf
+-- And also: https://themattchan.com/docs/algprog.pdf
+
+-- Guard function used in McCarthy's conditional
+-- | It provides information about the outcome of testing @p@ on some input @a@,
+-- encoded in terms of the coproduct injections without losing the input
+-- @a@ itself.
+grdS :: Selective f => f (a -> Bool) -> f a -> f (Either a a)
+grdS f a = (selector <$> (f <*> a)) <*> a
```

Comment:
I'm confused: this function doesn't seem to use the `Selective` interface, only `Applicative`. Is the implementation wrong, or the type signature?

Reply:
Yes, my bad. Thanks for noticing, the signature should use `Applicative` instead of `Selective`!